### DISCUSSION TITLE
Migração para outro framework JS

### DISCUSSION THREAD

**Discussion by igorpinhh** [frontendbr/forum#1123] [Perguntas e Respostas]
Na opinião de vocês uma pessoa que está saindo do AngularJS a fim de se atualizar, qual seria uma boa opção de framework para Spa's pensando no futuro? Ouvi falar mt bem do Angular 6 e React para projetos complexos, não achei posts muito conclusivos sobre ambos então queria uma opinião de quem usa no dia-a-dia...

**Comment by luizbills** [comment]
angular v1, você fala? Se for, recomendo o vue pois são meio parecidos.

**Comment by giovannism20** [comment]
Talvez no https://dev.to/ você encontre alguém falando sobre. Muita gente (pelo que eu vejo) concorda que aprender Vue seria melhor por conta mas ainda "sinto"/vejo que seria melhor o aprendizado do React por conta de vagas e afins.

**Comment by igorpinhh** [comment]
@luizbills sim angular v1, obrigado pela recomendação!

**Comment by igorpinhh** [comment]
@GiovanniSM20 valeu cara, vou dar uma pesquisada.

**Comment by giovannism20** [comment]
por nada @igorpinhh, abraço!

**Comment by rodrigooler** [comment]
@igorpinhh você pode dar uma experimentada no React sem medo algum.

Documentação: https://reactjs.org/
Hello World: https://reactjs.org/docs/hello-world.html

Você pode acompanhar também o uso continuo http://www.npmtrends.com/@angular/core-vs-angular-vs-react-vs-vue

**Comment by houstondapaz** [comment]
@igorpinhh como o @luizbills disse, o Vue é um bom ponto de partida, ele é bem proximo do angularjs mas traz tudo que vc precisa para aprender os novos frameworks, depois que aprendi Vue, React ficou bem mais simples de entender

**Comment by mahenrique94** [comment]
Todos os frameworks top do mercado como Angular, Vue e React vão lhe atender muito bem, porém, cada um com sua vantagem e custo.

Vue: Aprendizado muito fácil e simples, pois sabendo apenas HTML, CSS e JS você já consegue utilizá-lo.

React: Tem muito mercado e força, possuí as mesmas vantagens do Vue, com "HTML" (JSX), CSS e JS você já consegue começar a utilizá-lo.

Angular: Você vai precisar aprender e/ou saber TypeScript, se souber, show de bola, segue a mesma ideia de ambos acima.

Eu particularmente mexo e já mexi com os três, todos são muito bons e possuem praticamente os mesmos recursos e funcionalidades (cada um do seu jeito).

**Comment by ninetails** [comment]
A transição do AngularJS para os outros é: componentização. A idéia é a mesma para os 3, mas como falaram, o Vue é mais friendly para quem tá começando. Mas aprendeu componentização, o restante a diferença é como cada um é escrito e tem seus lifecycle (claro, falando de forma grosseira e superficial).

**Comment by waghcwb** [comment]
Estou testando `VueJS` em um projeto de `e-commerce` e recomendo. Curva de aprendizado curta e muito poderoso para construir interfaces reativas.
